### PR TITLE
docs(readme): Initial version of a readme

### PR DESCRIPTION
**Pull request by fadeev**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,31 +1,63 @@
-# spn
+# Starport Network
 
-**spn** is a blockchain application built using Cosmos SDK and Tendermint and generated with [Starport](https://github.com/tendermint/starport).
+Starport Network (SPN) is a network for launching [Cosmos](https://cosmos.network) blockchains by matching coordinators with validators. This repository contains an SPN node built with Cosmos SDK and generated with [Starport](https://github.com/tendermint/starport).
```
Comment (ltacker) [inline readme.md:3]: > This repository contains an SPN node

This sentence sounds like there are several spn nodes

File: readme.md [hunk 2]
```diff
@@ -1,31 +1,63 @@
...
+
+To initiate a blockchain launch run the following command:
+
+```
+starport network chain create --chain [chainID] --source [sourceURL]
+```
+
+`chainID` is a string that uniquely identifies your blockchain on SPN. `sourceURL` is a URL that can be used to clone the repository containing a Cosmos SDK blockchain node (for example, `https://github.com/tendermint/spn`). By running the `create` command you act as a "coordinator" and initiate the launch of a blockchain.
+
+By default a coordinator does not propose themselves as a validator. To do so, run `join` command and your proposal will be automatically approved.
```
Comment (ltacker) [inline readme.md:19]:
```suggestion
By default, a coordinator does not propose themselves as a validator. To do so, run `join` command and your proposals will be automatically approved.
```

Comment (ltacker) [inline readme.md:19]: The user may not know the notion of proposals and approvals at this point.
We may need a short introduction at the beginning of the readme.

- We record information of a blockchain to launch as a "chain"
- Users can send proposals to the chain. Proposals represent an additional piece for the launch of the blockchain (validator, account, genesis modification soon)
- Can be approved or rejected by the coordinator

We may also need to add some context on why we need this. Though we could put a link in the future (I think there will be a place in the starport documentation that contains a complete introduction to Starport Network)

Comment (ltacker) [inline readme.md:19]: I would put this statement below, in the joining as a validator section

File: readme.md (hunk 1 above)
Comment (ltacker) [inline readme.md:3]: I would put a link for the user to have more information about being a Validator

There is this one https://hub.cosmos.network/master/validators/validator-faq.html but it's related to Cosmos Hub, so I don't know if it makes sense to provide it

File: readme.md [hunk 3]
```diff
@@ -1,31 +1,63 @@
...
+starport network chain join [chainID]
 ```
-starport serve
+
+Follow the prompts to provide information about the validator. Starport will download the source code of the blockchain node, build, initialize and create and send two proposals to SPN: to add an account and to add a validator with self-delegation. By running a `join` command you act as a "validator".
+
+### Listing pending proposals
+
+```
+starport network proposal list
```
Comment (ltacker) [inline readme.md:34]:
```suggestion
starport network proposal list [chainID]
```

File: readme.md [hunk 4]
```diff
@@ -1,31 +1,63 @@
...
 ```
 
-`serve` command installs dependencies, initializes and runs the application.
+This command lists all pending proposals. To see accepted and rejected proposals, use `--accepted` and `--rejected` flags. Each proposal has a `proposalID` (integer, unique to the chain), this ID is used to approve and reject a proposal.
```
Comment (ltacker) [inline readme.md:37]: This command shows all the proposals by default now.
To filter status the flag is `--status [approved|pending|rejected]`

File: readme.md [hunk 5]
```diff
@@ -1,31 +1,63 @@
...
 
-## Configure
+### Accepting and rejecting proposals
 
-Initialization parameters of your app are stored in `config.yml`.
+As a coordinator run the following command to approve proposals:
 
-### `accounts`
+```
+starport network proposal approve 1,4,5,6
```
Comment (ltacker) [inline readme.md:44]:
```suggestion
starport network proposal approve [chainID] 1,4,5,6
```

File: readme.md (hunk 1 above)
Comment (fadeev) [reply readme.md]: A "validator" is a Tendermint concept, so we might as well link to the docs. Might not be the most gentle introduction, but it's technically correct (and should be good enough until we have a better article).

https://docs.tendermint.com/master/nodes/validators.html

Comment (ltacker) [reply readme.md]: > A "validator" is a Tendermint concept

Depends, you have the Tendermint validator, which is a node that proposes and validates blocks, and the Staking module validator which represents the account with delegated tokens.
With Starport Network want the coordinator to find Tendermint validators to make the chain working, but at the same time, the `AddValidator` proposal is related to the `staking` module so you would need some notion about this as well to understand staking and delegation.

But well, it's not relevant in the readme. That must be described in the docs

**Comment by ltacker** [comment]
That's a good initial version for the readme👍

One thing is that we will need to not forget updating it if `starport network` commands change

**Comment by fadeev** [comment]
Merging this. Improvements can be introduced in further PRs.

**Pull request merged**